### PR TITLE
Actually get real drag events on the map

### PR DESCRIPTION
**Pull request by cicloon**

### CODE REVIEW

**Comment by cicloon** [comment]
Updated

**Comment by larahoya** [comment]
🎉 🎉 🎉

File: src/android/plugin/google/maps/GoogleMaps.java [hunk 1]
```diff
@@ -1570,9 +1576,44 @@ public void onCameraChange(CameraPosition position) {
...
+    }
+
+    try {
+      params.put("reason", reasonStr);
+      jsonStr = params.toString();
+    } catch (JSONException e) {
+      e.printStackTrace();
+    }
+
+    webView.loadUrl("javascript:plugin.google.maps.Map._onCameraEvent('camera_move_started', " + jsonStr + ")");
```
Comment (lurecas) [inline src/android/plugin/google/maps/GoogleMaps.java:1608]: OMG... this plugin never ceases to amaze me...

File: src/android/plugin/google/maps/MyPluginLayout.java [hunk 2]
```diff
@@ -281,6 +285,50 @@ public void inValidate() {
     this.frontLayer.invalidate();
   }
 
+  public void setDragging() {
```
Comment (lurecas) [inline src/android/plugin/google/maps/MyPluginLayout.java]: Maybe this method can be called `setDraggingStarted` to indicate more clearly its intent

File: src/android/plugin/google/maps/MyPluginLayout.java [hunk 3]
```diff
@@ -281,6 +285,50 @@ public void inValidate() {
...
+    this.mapDragging = true;
+  }
+
+  public void clearDragging() {
```
Comment (lurecas) [inline src/android/plugin/google/maps/MyPluginLayout.java]: Maybe this method can be called `setDraggingEnded` to indicate more clearly its intent

**Comment by lurecas** [comment]
👌 🔝

File: src/android/plugin/google/maps/MyPluginLayout.java [hunk 4]
```diff
@@ -281,6 +285,50 @@ public void inValidate() {
...
+  public void setTouchYPosition(float y){
+    previousYPosition = y;
+  }
+
+
+  @Override
+  public boolean dispatchTouchEvent(MotionEvent event) {
+    int action = event.getAction();
+
+    if (mapDragging == true){
```
Comment (pedroduran) [inline src/android/plugin/google/maps/MyPluginLayout.java]: `if (mapDragging) {` does the trick

File: src/android/plugin/google/maps/MyPluginLayout.java (hunk 2 above)
Comment (pedroduran) [inline src/android/plugin/google/maps/MyPluginLayout.java]: Or `dragStarts()`

File: src/android/plugin/google/maps/MyPluginLayout.java (hunk 3 above)
Comment (pedroduran) [inline src/android/plugin/google/maps/MyPluginLayout.java]: and `dragEnds()`

File: src/android/plugin/google/maps/MyPluginLayout.java [hunk 5]
```diff
@@ -281,6 +285,50 @@ public void inValidate() {
...
 
+  public void setDragging() {
+    this.mapDragging = true;
+  }
+
+  public void clearDragging() {
+    this.mapDragging = false;
+  }
+
+  public void setTouchXPosition(float x){
```
Comment (pedroduran) [inline src/android/plugin/google/maps/MyPluginLayout.java]: If you need to call always both method together, it's more clear have just one public method `setTouchPosition(float x, float y)`

File: src/android/plugin/google/maps/MyPluginLayout.java (hunk 2 above)
Comment (pedroduran) [inline src/android/plugin/google/maps/MyPluginLayout.java]: definitivamente `dragStartsAt(float x, float y)` 💃

File: src/android/plugin/google/maps/MyPluginLayout.java [hunk 6]
```diff
@@ -332,6 +380,10 @@ public boolean onInterceptTouchEvent(MotionEvent event) {
         Log.e("client", "launching touch start to the JS" );
```
Comment (pedroduran) [inline src/android/plugin/google/maps/MyPluginLayout.java]: This log is an error?

File: src/android/plugin/google/maps/GoogleMaps.java [hunk 7]
```diff
@@ -1554,9 +1556,13 @@ public void onMapLoaded() {
    * Notify the myLocationChange event to JS
    */
   @Override
-  public void onCameraChange(CameraPosition position) {
+  public void onCameraIdle() {
+    CameraPosition position = map.getCameraPosition();
     JSONObject params = new JSONObject();
     String jsonStr = "";
+
+    Log.e("client", "onCameraIdle event");
```
Comment (pedroduran) [inline src/android/plugin/google/maps/GoogleMaps.java]: Error?

**Pull request merged**